s3/src/test/scala/org/apache/pekko/stream/connectors/s3/scaladsl/S3IntegrationSpec.scala

```diff
@@ -309,144 +460,161 @@ trait S3IntegrationSpec
-          case (versions, _) => versions
-        }, listBucketContentsAfterDelete)
+        } yield (versionsBeforeDelete.flatMap { case (versions, _) => versions },
+          versionsAfterDelete.flatMap {
+            case (versions, _) => versions
+          }, listBucketContentsAfterDelete)
 
-    val (versionsBeforeDelete, versionsAfterDelete, bucketContentsAfterDelete) = results.futureValue
+      val (versionsBeforeDelete, versionsAfterDelete, bucketContentsAfterDelete) = results.futureValue
 
-    versionsBeforeDelete.size shouldEqual 3
-    versionsAfterDelete.size shouldEqual 0
-    bucketContentsAfterDelete.size shouldEqual 0
+      versionsBeforeDelete.size shouldEqual 3
+      versionsAfterDelete.size shouldEqual 0
+      bucketContentsAfterDelete.size shouldEqual 0
   }
 
-  it should "listing object versions for a non versioned bucket should return None for versionId" in {
-    // TODO: Figure out a way to properly test this with Minio, see https://github.com/akka/alpakka/issues/2750
-    assume(this.isInstanceOf[AWSS3IntegrationSpec])
-    val objectKey = "listObjectVersionIdTest"
-    val bytes = ByteString(objectValue)
-    val data = Source.single(ByteString(objectValue))
-    val results = for {
-      _ <- S3
-        .putObject(defaultBucket,
-          objectKey,
-          data,
-          bytes.length,
-          s3Headers = S3Headers().withMetaHeaders(MetaHeaders(metaHeaders)))
-        .withAttributes(attributes)
-        .runWith(Sink.ignore)
-      result <- S3.listObjectVersions(defaultBucket, None).withAttributes(attributes).runWith(Sink.seq)
-      _ <- S3.deleteObject(defaultBucket, objectKey).withAttributes(attributes).runWith(Sink.ignore)
-    } yield result.flatMap { case (versions, _) => versions }
+  it should "listing object versions for a non versioned bucket should return None for versionId" in withDefaultBucket {
+    defaultBucket =>
+      // TODO: Figure out a way to properly test this with Minio, see https://github.com/akka/alpakka/issues/2750
+      assume(this.isInstanceOf[AWSS3IntegrationSpec])
+      val objectKey = "listObjectVersionIdTest"
+      val bytes = ByteString(objectValue)
+      val data = Source.single(ByteString(objectValue))
+      val results = for {
+        _ <- S3
+          .putObject(defaultBucket,
+            objectKey,
+            data,
+            bytes.length,
+            s3Headers = S3Headers().withMetaHeaders(MetaHeaders(metaHeaders)))
+          .withAttributes(attributes)
+          .runWith(Sink.ignore)
+        result <- S3.listObjectVersions(defaultBucket, None).withAttributes(attributes).runWith(Sink.seq)
+        _ <- S3.deleteObject(defaultBucket, objectKey).withAttributes(attributes).runWith(Sink.ignore)
+      } yield result.flatMap { case (versions, _) => versions }
 
-    forEvery(results.futureValue) { version =>
-      version.versionId shouldEqual None
-    }
+      Inspectors.forEvery(results.futureValue) { version =>
```

Comment:
This is because the `forEvery` from Scalatest conflicts with the `forEvery` from scalacheck (which is being introduced in this PR).